formats/src/main/scala/com/gu/scanamo/DynamoFormat.scala

```diff
@@ -309,71 +310,102 @@ object DynamoFormat extends EnumDynamoFormat {
       _.map(f.write).toList.asJava
     )(javaListFormat)
 
-  private val javaNumSetFormat = attribute(_.getNS, "NS")(_.withNS)
-  private val javaStringSetFormat = attribute(_.getSS, "SS")(_.withSS)
-  private def setFormat[T](r: String => Either[DynamoReadError, T])(w: T => String)(
-      df: DynamoFormat[java.util.List[String]]): DynamoFormat[Set[T]] =
-    xmap[Set[T], java.util.List[String]](_.asScala.toList.traverse(r).map(_.toSet))(
-      _.map(w).toList.asJava
-    )(df)
+  private def numSetFormat[T](r: String => Either[DynamoReadError, T])(w: T => String): DynamoFormat[Set[T]] =
+    new DynamoFormat[Set[T]] {
+      override def read(av: AttributeValue): Either[DynamoReadError, Set[T]] =
+        for {
+          ss <- Either.fromOption(Option(av.getSS), NoPropertyOfType("NS", av))
```

Comment:
Good spot! Turns out I managed to break the tests, which was hiding the fact that I wired this in wrong. Should have been `getNS` instead of `getSS` etc.

See: https://github.com/scanamo/scanamo/pull/250/commits/bab4151900ae6c724e491e541d9df8d9d1031555